File: android/src/main/kotlin/com/plugin/mapboxnav/domain/models/NavigationBehaviorPolicy.kt
package com.plugin.mapboxnav.domain.models

data class NavigationBehaviorPolicy(
    val autoOverviewOnRouteReady: Boolean = true,
    val autoFollowOnFirstLocation: Boolean = true,
    val autoFollowOnDestinationChange: Boolean = true,
    val autoStartTripSession: Boolean = true,
) {
    companion object {
        fun fromMap(data: Map<String, Any>?): NavigationBehaviorPolicy {
            if (data == null) return NavigationBehaviorPolicy()
            return NavigationBehaviorPolicy(
                autoOverviewOnRouteReady = data["autoOverviewOnRouteReady"] as? Boolean ?: true,
                autoFollowOnFirstLocation = data["autoFollowOnFirstLocation"] as? Boolean ?: true,
                autoFollowOnDestinationChange = data["autoFollowOnDestinationChange"] as? Boolean ?: true,
                autoStartTripSession = data["autoStartTripSession"] as? Boolean ?: true,
            )
        }
    }
}

File: lib/src/data/navigation_behavior_policy.dart
class NavigationBehaviorPolicy {
  final bool autoOverviewOnRouteReady;
  final bool autoFollowOnFirstLocation;
  final bool autoFollowOnDestinationChange;
  final bool autoStartTripSession;

  const NavigationBehaviorPolicy({
    this.autoOverviewOnRouteReady = true,
    this.autoFollowOnFirstLocation = true,
    this.autoFollowOnDestinationChange = true,
    this.autoStartTripSession = true,
  });

  Map<String, dynamic> toMap() => {
        'autoOverviewOnRouteReady': autoOverviewOnRouteReady,
        'autoFollowOnFirstLocation': autoFollowOnFirstLocation,
        'autoFollowOnDestinationChange': autoFollowOnDestinationChange,
        'autoStartTripSession': autoStartTripSession,
      };

  factory NavigationBehaviorPolicy.defaults() =>
      const NavigationBehaviorPolicy();
}

